device/colibri.go

```diff
@@ -0,0 +1,314 @@
+func (d *colibri) Prepare() error {
+	// start VM, upload image and extract it
+	d.flasher.Prepare()
+	log.WithField("device", COLIBRI).Debug("Prepare")
+	// install toradex flasher dependencies
+	d.installTools()
+	return nil
+}
+
+func (d *colibri) getPort() error {
+	if d.Port != "" {
+		return nil
+	}
+	var perr error
+	for attempt := 0; attempt < portSelectionTries; attempt++ {
+		if attempt > 0 && !dialogs.YesNoDialog("No ports found. Reconnect your device and try again. Ready?") {
+			return perr
+		}
+		fmt.Println("[+] Enumerating serial ports...")
+		port, err := serialport.GetPort("auto")
+		if err != nil {
+			perr = err
+			continue
+		}
+		d.Port = port
+		break
+	}
+	if d.Port == "" {
+		return perr
+	}
+	fmt.Println("[+] Using ", dialogs.PrintColored(d.Port))
+	return nil
+}
+
+// Configure overrides flasher Configure() method with custom image configuration
+func (d *colibri) Configure() error {
+	log.WithField("device", COLIBRI).Debug("Configure")
+	fmt.Println("[+] Configuring...")
+
+	job := help.NewBackgroundJob()
+
+	go func() {
+		defer job.Close()
+
+		if err := d.configureImage(); err != nil {
+			job.Error(err)
+			return
+		}
+	}()
+
+	if err := help.WaitJobAndSpin("Preparing", job); err != nil {
+		return err
+	}
+
+	fmt.Println("[+] Image ready")
+	return nil
+}
+
+// Flash configures and flashes image
+func (d *colibri) Flash() error {
+	if err := d.Prepare(); err != nil {
+		return err
+	}
+
+	if err := d.Configure(); err != nil {
+		return err
+	}
+
+	if err := d.Write(); err != nil {
+		return err
+	}
+
+	return d.Done()
+}
+
+func (d *colibri) installTools() error {
+	if err := d.exec("apk add dosfstools parted sudo e2fsprogs-extra coreutils libattr zip"); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (d *colibri) configureImage() error {
+	d.img = "colibri_image.zip"
+
+	log.Debug("Creating tmp folder")
+	command := fmt.Sprintf("mkdir -p %s", config.MountDir)
+	if err := d.exec(command); err != nil {
+		return err
+	}
+	log.Debug("Running update.sh")
+	command = fmt.Sprintf("cd %s && ./update.sh -o %s", help.AddPathSuffix("unix", config.TmpDir, d.folder), config.MountDir)
+	if err := d.exec(command); err != nil {
+		return err
+	}
+	command = fmt.Sprintf("cd %s && zip -0 -r ../%s *", config.MountDir, d.img)
+	if err := d.exec(command); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Write - writes image to SD card
+func (d *colibri) Write() error {
+	if d.img != "" {
+		log.WithField("img", d.img).Debug("Downloading image from vbox")
+
+		job := help.NewBackgroundJob()
+		go func() {
+			defer job.Close()
+			if err := d.conf.SSH.ScpFrom(help.AddPathSuffix("unix", config.TmpDir, d.img), filepath.Join(help.GetTempDir(), d.img)); err != nil {
+				job.Error(err)
+			}
+		}()
+
+		if err := help.WaitJobAndSpin("Copying files", job); err != nil {
+			log.Error(err)
+			return err
+		}
+	} else {
+		d.img = "colibri_image.zip"
+	}
+
+	flash_only := d.flasher.CLI.Bool("flash")
```

Comment:
__[golint]__
* [golint] don't use underscores in Go names; var flash_only should be flashOnly

<!-- Comment by [SideCI](https://sideci.com) -->